zerocopy-derive/src/lib.rs

```diff
@@ -548,11 +578,19 @@ fn impl_block<D: DataExt>(
     let trait_ident = trt.ident();
     let field_types = data.field_types();
 
-    let field_type_bounds = require_trait_bound_on_field_types
-        .then(|| field_types.iter().map(|ty| parse_quote!(#ty: ::zerocopy::#trait_ident)))
-        .into_iter()
-        .flatten()
-        .collect::<Vec<_>>();
+    let field_type_bounds: Vec<_> = match require_trait_bound_on_field_types {
+        RequireBoundedFields::No => vec![],
+        RequireBoundedFields::Yes => {
+            field_types.iter().map(|ty| parse_quote!(#ty: ::zerocopy::#trait_ident)).collect()
+        }
+        RequireBoundedFields::Trailing => {
+            if let Some(trailing) = field_types.last() {
+                vec![parse_quote!(#trailing: ::zerocopy::#trait_ident)]
+            } else {
+                vec![]
+            }
+        }
+    };
```

Comment:
If you want to be really cute. I swapped the order of `Yes` and `No` so that the two `Trailing` branches are grouped. Whether this is a good idea is highly dependent upon personal preference. Feel free to take all, some (ie, just the `bound_tt` idea), or none of this suggestion.

```suggestion
    let bound_tt = |ty| parse_quote!(#ty: ::zerocopy::#trait_ident);
    let field_type_bounds: Vec<_> = match (require_trait_bound_on_field_types, field_types) {
        (RequireBoundedFields::Yes, _) => field_types.iter().map(bound_tt).collect(),
        (RequireBoundedFields::No, _) | (RequireBoundedFields::Trailing, []) => vec![],
        (RequireBoundedFields::Trailing, [.., last]) => vec![bound_tt(last)],
    };
```

Reply:
I *do* love being cute.